src/pages/calculator.tsx

```diff
@@ -0,0 +1,328 @@
+import FinalAmount from '@components/calculator/FinalAmount';
+import CTA from '@components/CTA';
+import {
+	FormInput,
+	FormInputWrapper,
+	FormRadioGroup,
+	FormRadioLabel,
+	FormRadioWrapper,
+	FormSelect,
+} from '@components/styled/Forms';
+import Link from 'next/link';
+import { useState } from 'react';
+import styled from 'styled-components';
+
+type OnChangeType =
+	| React.ChangeEvent<HTMLInputElement>
+	| React.ChangeEvent<HTMLSelectElement>;
+
+type FormDataType = {
+	shape: string;
+	diameter: string;
+	length: string;
+	width: string;
+	thickness: string;
+	unit: string;
+};
+
+const dataOne = {
+	productType: 'OEX5302',
+	ratio: 2,
+	partA: 40,
+	partB: 10,
+};
+const dataTwo = {
+	productType: 'OEX5311',
+	ratio: 1,
+	partA: 30,
+	partB: 30,
+};
+
+function Calculator() {
+	const [formData, setFormData] = useState<FormDataType>({
+		shape: 'cylinder',
+		diameter: '',
+		length: '',
+		width: '',
+		thickness: '',
+		unit: '',
+	});
+
+	const { shape, length, width, thickness, diameter } = formData;
+
+	const onInputChange = (e: OnChangeType) => {
+		setFormData(prev => {
+			return { ...prev, [e.target.id]: e.target.value };
+		});
+	};
+
+	return (
+		<>
+			<PageHeading>
+				<BackButtonWrapper>
+					<Link href="/composites">Back</Link>
+				</BackButtonWrapper>
+				<PageHeadingWrapper>
+					<span>Calculator</span>
+				</PageHeadingWrapper>
+			</PageHeading>
+			<PageWrapper>
+				<PageContainer>
+					<CalculatorWrapper>
+						<CalculateContent>
+							<h3>Epoxy Calculator</h3>
+							<p>
+								Leave out the guesswork. Use our epoxy resin to estimate the
+								amount of resin you would need for your project!
+							</p>
+							<form action="">
+								<ParagraphText>
+									What is the shape of your project?
+								</ParagraphText>
+								<FormRadioWrapper>
+									<FormRadioGroup>
+										<FormRadioLabel>
+											<input
+												type="radio"
+												id="shape"
```

Comment:
your ids should be unique, compare them with ln100. That's the key for your `onInputChange` above

Reply:
What is In100? Should the id be something like shape100?

Reply:
If the shape has a different id, i will use the name as the key, for radio buttons to work together, they must have the same name attribute

Reply:
line 100

Reply:
consider using the name in your onChange then, instead of ID, so that ID would be unique in a page